plugins/become/ksusu.py

```diff
@@ -0,0 +1,182 @@
+                key: flags_ksu
+            vars:
+              - name: ansible_become_flags_ksu
+              - name: ansible_ksusu_flags_ksu
+            env:
+              - name: ANSIBLE_BECOME_FLAGS_KSU
+              - name: ANSIBLE_KSU_FLAGS_KSU
+        become_flags:
+            description: Options to pass to su
+            default: ''
+            ini:
+              - section: privilege_escalation
+                key: become_flags
+              - section: privilege_escalation
+                key: become_flags_su
+              - section: ksusu_become_plugin
+                key: flags_su
+            vars:
+              - name: ansible_become_flags
+              - name: ansible_become_flags_su
+              - name: ansible_ksusu_flags
+              - name: ansible_ksusu_flags_su
+            env:
+              - name: ANSIBLE_BECOME_FLAGS
+              - name: ANSIBLE_BECOME_FLAGS_SU
+              - name: ANSIBLE_KSU_FLAGS_SU
+        become_pass:
+            description: ksu password
+            required: False
+            vars:
+              - name: ansible_ksusu_pass
+              - name: ansible_become_pass
+              - name: ansible_become_password
+            env:
+              - name: ANSIBLE_BECOME_PASS
+              - name: ANSIBLE_KSUSU_PASS
+            ini:
+              - section: ksusu_become_plugin
+                key: password
+        prompt_l10n:
+            description:
+                - List of localized strings to match for prompt detection
+                - If empty we'll use the built in one
+            default: []
+            ini:
+              - section: ksusu_become_plugin
+                key: localized_prompts
+            vars:
+              - name: ansible_ksusu_prompt_l10n
+            env:
+              - name: ANSIBLE_KSUSU_PROMPT_L10N
+"""
+
+import re
+
+from ansible.module_utils.common.text.converters import to_bytes
+from ansible.plugins.become import BecomeBase
+
+
+class BecomeModule(BecomeBase):
+
+    name = 'community.general.ksusu'
+
+    # messages for detecting prompted password issues
+    fail = ('Password incorrect',)
+    missing = ('No password given',)
+
+    def check_password_prompt(self, b_output):
+        ''' checks if the expected password prompt exists in b_output '''
+
+        prompts = self.get_option('prompt_l10n') or ["Kerberos password for .*@.*:"]
+        b_prompt = b"|".join(to_bytes(p) for p in prompts)
+
+        return bool(re.match(b_prompt, b_output))
+
+    def build_become_command(self, cmd, shell):
+
+        super(BecomeModule, self).build_become_command(cmd, shell)
+
+        # Prompt handling for ``ksu`` is more complicated, this
+        # is used to satisfy the connection plugin
+        self.prompt = True
+
+        if not cmd:
+            return cmd
+
+        exe_ksu = self.get_option('become_exe_ksu')
+        exe_su = self.get_option('become_exe_su') or '/bin/su'  # Must be absolute
```

Comment:
Why use `or ...` here (and below)?

Reply:
Python is not my native language;) The code is trying to mimic the style used in the become ksu and sudosu (which this is really just a mash up of the two). I'm happy with whatever your preference is here.

FYI: as hopefully clear in the documentation, this has to be an absolute path as ksu -e requires such.

Reply:
I checked and at least in stable-2.14 get_option doesn't support a default.

Reply:
`get_option` always supported defaults. The default for `become_exe_su` is in line 69 (https://github.com/ansible-collections/community.general/pull/5696/files#diff-0bbe34c88f6040f50adfa1bc8e26ac352c4a3dac565c437e06c7dda23b63e12bR69).

IIRC the need to list the defaults again came for Ansible 2.9, and/or the unit tests. It should no longer be necessary.

Reply:
(Actually if you check the other existing become plugins, you can see that they are no longer doing this.)

Reply:
Ok, so I'm confused:

Are you saying there should documentation of the default? I will documentation of the default.

Are you saying there should be a "or 'ksu'" added to line 174?

Are you saying that get_option will pull a default from the documentation and I do not need to provide the default via 'or' here? I've gone through other become plugins in both the devel branch of ansible and main of community_general and all the usage of get_option I'm seeing are get_option(...) or default.

I was thinking something like a default argument for get_option which I am not seeing.

Reply:
No, I'm saying that the documentated default **is** the default for plugins (when the configuration manager is used). When you run `self.get_option('become_exe_su')`, it will return `/bin/su` if the user did not configure it explicitly (by using any of the documented ways to configure it in https://github.com/ansible-collections/community.general/pull/5696/files#diff-0bbe34c88f6040f50adfa1bc8e26ac352c4a3dac565c437e06c7dda23b63e12bR70-R81).

Reply:
I believe I have made the change you desire.

Could you please point me to the API for ansible (eg: similar to javadoc or man pages on each interface)? I am not seeing any doc strings for the different arguments to this core object/method. 

Note that the become plugins in ansible itself (su & sudo), both 2.14 and devel, do this with 'or' instead of this feature.